Library/MapleBacon/MapleBacon/ImageGifExtension.swift

```diff
@@ -45,30 +45,37 @@ extension UIImage {
         return frames
     }
 
-    private class func createImagesAndDelays(source: CGImageSourceRef) -> ([CGImageRef], [Int]) {
+    private class func createImagesAndDelays(source: CGImageSource) -> ([CGImage], [Int]) {
         let count = Int(CGImageSourceGetCount(source))
-        var images = [CGImageRef]()
+        var images = [CGImage]()
         var delayCentiseconds = [Int]()
         for i in 0 ..< count {
             if let image = CGImageSourceCreateImageAtIndex(source, i, nil) {
                 images.append(image)
-                delayCentiseconds.append(delayCentisecondsForImageAtIndex(source, index: i))
+                delayCentiseconds.append(delayCentisecondsForImageAtIndex(source: source, index: i))
             }
         }
         return (images, delayCentiseconds)
     }
 
-    private class func delayCentisecondsForImageAtIndex(let source: CGImageSourceRef, let index: Int) -> Int {
-        if let cfProperties = CGImageSourceCopyPropertiesAtIndex(source, index, nil),
-        properties: NSDictionary = cfProperties,
-            gifProperties = properties[kCGImagePropertyGIFDictionary as String] as? NSDictionary {
-            var delayTime = gifProperties[kCGImagePropertyGIFUnclampedDelayTime as String] as! NSNumber
-            if delayTime.doubleValue == 0 {
-                delayTime = gifProperties[kCGImagePropertyGIFDelayTime as String] as! NSNumber
+    private class func delayCentisecondsForImageAtIndex(source: CGImageSource, index: Int) -> Int {
+        
+        guard
```

Comment:
just single `let` in the guard statement. And no need for the type declarations either

```swift
guard let properties = …, gifProperties = …, var delayTime = … else {}`
```